### PR TITLE
Extend puppet type check for install_method

### PR DESCRIPTION
**Pull request by bastelfreak**
previously we had a case statement that checked for 3 values in a
string. We now uplift those 3 values into an enum to make validation
better and easier to test.

### CODE REVIEW

**Comment by bastelfreak** [comment]
I did not align the whitespaces here because that just creates a huge huge diff without any benefit.

**Comment by igalic** [comment]
> I did not align the whitespaces here because that just creates a huge huge diff without any benefit.

until we find a better waytto make unaligned puppet code more readable, we should do alignment fixes in a separate commit

**Comment by bastelfreak** [comment]
okay, will do

**Comment by bastelfreak** [comment]
instead of realigning everything I moved the enum into a custom type. benefit: we can now update the enum at a single place to adjust it for every of our exporters \o/

File: REFERENCE.md [hunk 1]
```diff
@@ -1200,6 +1209,14 @@ Data type: `Boolean`
 
 Default value: ``false``
 
+##### `scrape_host`
+
+Data type: `Optional[Stdlib::Host]`
+
+
```
Comment (igalic) [inline REFERENCE.md:1216]: this is fairly sparse on documentation

**Review by igalic — DISMISSED** [review]
i'd like to see less missing documentation

File: REFERENCE.md (hunk 1 above)
Comment (bastelfreak) [reply REFERENCE.md:1216]: but isn't that out of scope for this PR?

Comment (igalic) [reply REFERENCE.md:1216]: fair

File: manifests/daemon.pp [hunk 2]
```diff
@@ -129,9 +129,7 @@
       }
     }
     'none': {}
-    default: {
-      fail("The provided install method ${install_method} is invalid")
-    }
+    default: {}
```
Comment (igalic) [inline manifests/daemon.pp:132]: do we still need a default case, if we exhaust all enum options?

**Review by igalic — COMMENTED** [review]
🙋🏻‍♀️

**Review by igalic — APPROVED** [review]

File: manifests/daemon.pp (hunk 2 above)
Comment (bastelfreak) [reply manifests/daemon.pp:132]: in theory not. but puppet-lint demands one. If we remove it we need to add a comment to disable the puppet-lint check :(

Comment (igalic) [reply manifests/daemon.pp:132]: sounds like a bug in puppet lint

**Pull request merged**